### ISSUE TITLE
Connection impossible if the `mosh-server` command does not terminate

### ISSUE THREAD

**Issue by panicbit** [mobile-shell/mosh#814]
If `mosh-server` command does not terminate and the SSH session stays open, `mosh` won't progress. As a result the server will time out, after which the client gets started and times out as well.

For some reason JuiceSSH is not affected by this (it probably starts the client as soon as it sees the `CONNECT` line).

I was facing this situation because I was running `mosh-server` under `proot`, which only terminates when all of its child processes terminate. After identifying this issue I could work around it with a wrapper:

```bash
exec 3< <(YOUR_MOSH_SERVER_CMD "$@")
grep -m 1 CONNECT <&3
```

This looks like a wontfix to me, but it should be useful info for other people having similarly excotic environments :smile:.

**Comment by cgull** [comment]
This is really more of an issue with `proot` than with `mosh`, but it does expose an interesting point.  Currently, the `mosh` script waits for its input from `ssh` to close before starting `mosh-server`, even if it's gotten all the information it needs.  This could be changed, `mosh` could stop looking for input from its children once it has received `MOSH IP` and `MOSH CONNECT`.

A workaround would be to add something like this to the pipe-read loop in `mosh`:
```perl
  last if defined $ip and defined $key;
```
Issues: this would cause mosh.pl to drop any extra/unexpected/error output from the server, and @panicbit thinks it might kill `proot` by closing the ssh session it's on.

**Comment by cgull** [comment]
#833 may be related to this too.
